projects/uswds-components/src/lib/search/search.component.ts

```diff
@@ -0,0 +1,101 @@
+import { ChangeDetectionStrategy, ChangeDetectorRef, Component, ElementRef, EventEmitter, forwardRef, Input, OnInit, Output, ViewChild } from '@angular/core';
+import { NG_VALUE_ACCESSOR, ControlValueAccessor } from '@angular/forms';
+
+// Search Settings class
+export class SearchSettings {
+  public ariaLabel: string;
+  public id: string;
+  public size: string;
+  public buttonText: string;
+  public placeholder: string
+}
+
+@Component({
+  selector: 'usa-search',
+  templateUrl: './search.component.html',
+  providers: [
+    {
+      provide: NG_VALUE_ACCESSOR,
+      useExisting: forwardRef(() => SearchComponent),
+      multi: true,
+    },
+  ],
+  changeDetection: ChangeDetectionStrategy.OnPush,
+})
+export class SearchComponent {
+
+  model: string = '';
+
+  private _onChange = (_: any) => { };
+  private _onTouched = () => { };
+
+  @Input() searchSettings: SearchSettings = new SearchSettings();
+
+  @Output() onBlurChange: EventEmitter<string> = new EventEmitter(null);
+
+  @Output() onSearchTextChange: EventEmitter<string> = new EventEmitter(null);
+
+  constructor(public cdr: ChangeDetectorRef) { }
+
+  focusChange(event) {
+    this.onBlurChange.emit(event.target.value)
+  }
+
+  onKeydown(event): void {
+    if (event.code == 'Enter') {
```

Comment:
Could we use an enum here instead, you'll find Enum values for keyboard here - https://github.com/GSA/ngx-uswds/blob/main/projects/uswds-components/src/lib/util/key.ts